widget/list.go

```diff
@@ -0,0 +1,449 @@
+		if l.list.selectedItem != nil && l.list.selectedIndex >= l.firstItemIndex && l.list.selectedIndex <= l.lastItemIndex {
+			l.list.selectedItem.selected = false
+			l.list.selectedItem.Refresh()
+		}
+		l.list.selectedItem = item.(*listItem)
+		l.list.selectedIndex = index
+		l.list.selectedItem.selected = true
+		l.list.selectedItem.Refresh()
+		if l.list.OnItemSelected != nil {
+			l.list.OnItemSelected(index, item.(*listItem).child)
+		}
+	}
+}
+
+type listItem struct {
+	BaseWidget
+
+	onTapped          func()
+	statusIndicator   *canvas.Rectangle
+	child             fyne.CanvasObject
+	divider           *canvas.Rectangle
+	hovered, selected bool
+}
+
+func newListItem(child fyne.CanvasObject, tapped func()) *listItem {
+	li := &listItem{
+		child:    child,
+		onTapped: tapped,
+	}
+
+	li.ExtendBaseWidget(li)
+	return li
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to its renderer
+func (li *listItem) CreateRenderer() fyne.WidgetRenderer {
+	li.ExtendBaseWidget(li)
+
+	li.statusIndicator = canvas.NewRectangle(theme.BackgroundColor())
+	li.divider = canvas.NewRectangle(theme.ShadowColor())
+
+	objects := []fyne.CanvasObject{li.statusIndicator, li.child, li.divider}
+
+	return &listItemRenderer{widget.NewBaseRenderer(objects), li}
+}
+
+// MinSize returns the size that this widget should not shrink below
+func (li *listItem) MinSize() fyne.Size {
+	li.ExtendBaseWidget(li)
+	return li.BaseWidget.MinSize()
+}
+
+// MouseIn is called when a desktop pointer enters the widget
+func (li *listItem) MouseIn(*desktop.MouseEvent) {
+	li.hovered = true
+	li.Refresh()
+}
+
+// MouseMoved is called when a desktop pointer hovers over the widget
+func (li *listItem) MouseMoved(*desktop.MouseEvent) {
+}
+
+// MouseOut is called when a desktop pointer exits the widget
+func (li *listItem) MouseOut() {
+	li.hovered = false
+	li.Refresh()
+}
+
+// Tapped is called when a pointer tapped event is captured and triggers any tap handler
+func (li *listItem) Tapped(*fyne.PointEvent) {
+	if li.onTapped != nil {
+		li.selected = true
+		li.Refresh()
+		li.onTapped()
+	}
+}
+
+type listItemRenderer struct {
+	widget.BaseRenderer
+
+	item *listItem
+}
+
+// MinSize calculates the minimum size of a listItem.
+// This is based on the size of the status indicator and the size of the child object.
+func (li *listItemRenderer) MinSize() (size fyne.Size) {
+	size = fyne.NewSize((theme.Padding()*2)+li.item.child.Size().Width,
+		li.item.child.Size().Height+theme.Padding()*2)
+	return
+}
+
+// Layout the components of the listItem widget
+func (li *listItemRenderer) Layout(size fyne.Size) {
+	li.item.statusIndicator.Move(fyne.NewPos(0, 0))
+	s := fyne.NewSize(theme.Padding(), size.Height-1)
+	li.item.statusIndicator.SetMinSize(s)
+	li.item.statusIndicator.Resize(s)
+
+	x := theme.Padding() * 2
+	li.item.child.Move(fyne.NewPos(x, theme.Padding()))
+	li.item.child.Resize(fyne.NewSize(size.Width-x, size.Height-(2*theme.Padding())))
```

Comment:
How much padding does this leave between the trailing edge of the child, and the trailing edge of the list?

Reply:
2 * theme.Padding

Reply:
actually none.  This padding is between the child provided by the dev and the trailing edge of our list item.  There is no padding between the trailing edge of our list item and the list.